### DISCUSSION TITLE
Support for mixed discrete-continuous actions

### DISCUSSION THREAD

**Discussion by CloudyDory** [thu-ml/tianshou#856] [Q&A]
Hi, I have a custom environment that needs to input both discrete and continuous actions. The continuous action should be a Gaussian policy with learned mean and standard deviation. I hope to train a PPO agent to solve the task. Is it still possible to use Tianshou's default Collector and OnPolicyTrainer, or should I write my custom Collector and OnPolicyTrainer?

**Comment by Trinkle23897** [comment]
imo you only need to inherit PPOPolicy and rewrite parts of them (forward, learn) to let it support combined action input. No need to rewrite Collector and Trainer

**Reply by CloudyDory** [reply]
Hi, I guess the collector should also be rewritten as well?

**Reply by Trinkle23897** [reply]
I don't think so. You can pack your action as a dict (similar to dict action space) each time, which will be converted to `Batch` automatically in `Collector`.
